CreatubblesAPIClient/Sources/CreationUploadSession/CreationUploadSession.swift

```diff
@@ -370,15 +369,14 @@ class CreationUploadSession: NSObject, Cancelable
         return paths.first!
     }
     
-    fileprivate func saveCurrentImage(_ completion: (Error?) -> Void)
+    fileprivate func storeCreation(_ completion: ((Error?) -> Void) )
     {
-        if(creationData.dataType != .image)
-        {
-            completion(nil)
-            return
-        }
+        var data: Data!
+        
+        if let creationData = self.creationData.data  { data = creationData }
+        else if let image   = self.creationData.image { data = UIImageJPEGRepresentation(image, 1)! }
+        else if let url     = self.creationData.url   { data = try? Data(contentsOf: url) }
         
-        let data = UIImageJPEGRepresentation(creationData.image!, 1)!
         let url = URL(fileURLWithPath: (CreationUploadSession.documentsDirectory()+"/"+relativeImageFilePath))
```

Comment:
I would change `relativeImageFilePath` to `relativeFilePath` - in my opinion it is confusing.